### ISSUE TITLE
Transactional Emails Subjectline - save removes Unicode EMOJIs...

### ISSUE THREAD

**Issue by tomekjordan** [OpenMage/magento-lts#430]
Hi. I tried every possible method to include Emoji in the transactional emails, but looks like code is cut/changed after email template save action.
This blocks merchants from using Emojis in the emails.
Emojis cane be only saved with format like: &#127872; {{var customer_name}}
and grid shows them correctly [http://prntscr.com/i0a30d](http://prntscr.com/i0a30d)
but Magento email sending makes them unreadable and shows accidental chars in the inboxes

Any options to fix it and allow using/showing UNICODE in transactional emails?

Thanks

**Comment by LeeSaferite** [comment]
First, before we check anything else, figure out if your emoji use 3 or 4 bytes. I alsk because the DB is likely (99.99999999%) set up using "utf8" which in the weird MySQL world really means 3-byte UTF8. To get true 4-byte support, the table column needs to use "utf8mb4". Look it up, you'll see what I mean.

**Comment by luigifab** [comment]
That's true, at minimum the database column must use _utf8mb4_ charset (for example _utf8mb4_general_ci_). Moreover the associated model must set and unset _SET NAMES utf8mb4_ on __beforeLoad_ and __afterLoad_.

But, to extend that everywhere, is a good idea to convert database and all tables/columns to _utf8mb4_ instead of _utf8_?

**Comment by tmotyl** [comment]
Beware that converting everything to utf8mb4 can cause issues witg too long indexes and so (also different versions of mysql behaves differently) so its not an quick and easy change

**Comment by colinmollenhour** [comment]
Since `utf8` might be a subset of `utf8mb4` would it be possible to change the init statement to `SET NAMES utf8mb4 COLLATE utf8mb4_unicode_ci;` and then only change the specific columns that need full UTF-8 support? For the database there is nothing saying that all columns and tables have to have the same encoding.

I have a script that generates alter statements to just do a blanket update if you're interested.

**Comment by luigifab** [comment]
We have applied the change in local.xml and in some database columns, and it is perfect.

**Comment by m-overlund** [comment]
Maybe useful dev article, regarding caveats when changing from utf8 to utf8mb4
https://dev.relatel.dk/blog/quick-no-hack-emoji-support-with-mysql-rails/
